frame/support/procedural/src/pallet/parse/config.rs

```diff
@@ -0,0 +1,385 @@
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::helper;
+use syn::spanned::Spanned;
+use quote::ToTokens;
+
+/// List of additional token to be used for parsing.
+mod keyword {
+	syn::custom_keyword!(Config);
+	syn::custom_keyword!(From);
+	syn::custom_keyword!(T);
+	syn::custom_keyword!(I);
+	syn::custom_keyword!(Get);
+	syn::custom_keyword!(config);
+	syn::custom_keyword!(IsType);
+	syn::custom_keyword!(Event);
+	syn::custom_keyword!(constant);
+	syn::custom_keyword!(frame_system);
+	syn::custom_keyword!(disable_frame_system_supertrait_check);
+}
+
+/// Input definition for the pallet config.
+pub struct ConfigDef {
+	/// The index of item in pallet module.
+	pub index: usize,
+	/// Whether the trait has instance (i.e. define with `Config<I = ()>`)
+	pub has_instance: bool,
+	/// Const associated type.
+	pub consts_metadata: Vec<ConstMetadataDef>,
+	/// Whether the trait has the associated type `Event`, note that those bounds are checked:
+	/// * `IsType<Self as frame_system::Config>::Event`
+	/// * `From<Event>` or `From<Event<T>>` or `From<Event<T, I>>`
+	pub has_event_type: bool,
+	/// The where clause on trait definition but modified so `Self` is `T`.
+	pub where_clause: Option<syn::WhereClause>,
+
+}
+
+/// Input definition for a constant in pallet config.
+pub struct ConstMetadataDef {
+	/// Name of the associated type.
+	pub ident: syn::Ident,
+	/// The type in Get, e.g. `u32` in `type Foo: Get<u32>;`, but `Self` is replaced by `T`
+	pub type_: syn::Type,
+	/// The doc associated
+	pub doc: Vec<syn::Lit>,
+}
+
+impl syn::parse::Parse for ConstMetadataDef  {
+	fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+		let doc = helper::get_doc_literals(&syn::Attribute::parse_outer(input)?);
+		input.parse::<syn::Token![type]>()?;
+		let ident = input.parse::<syn::Ident>()?;
+		input.parse::<syn::Token![:]>()?;
+		input.parse::<keyword::Get>()?;
+		input.parse::<syn::Token![<]>()?;
+		let mut type_ = input.parse::<syn::Type>()?;
+		type_ = syn::parse2::<syn::Type>(replace_self_by_t(type_.to_token_stream()))
+			.expect("Internal error: replacing `Self` by `T` should result in valid type");
+		input.parse::<syn::Token![>]>()?;
+		input.parse::<syn::Token![;]>()?;
+
+		Ok(Self { ident, type_, doc })
+	}
+}
+
+/// Parse for `#[pallet::disable_frame_system_supertrait_check]`
+pub struct DisableFrameSystemSupertraitCheck;
```

Comment:
What is the usage of this? I haven't seen anything about it anywhere in the docs.

Reply:
the doc quickly explains it:

> To bypass the frame_system::Config supertrait check, use the attribute #[pallet::disable_frame_system_supertrait_check], e.g.: $example_inserted_here

But I never emphasized it as it rather useless.

The only useful usage of it is to be able to define frame_system pallet using this macro